streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -342,4 +342,20 @@ private RecordQueue createRecordQueue(TopicPartition partition, SourceNode sourc
         }
     }
 
+    @SuppressWarnings("unchecked")
+    public <K, V> void forward(K key, V value, String childName) {
+        ProcessorNode thisNode = currNode;
+        try {
+            for (ProcessorNode childNode : (List<ProcessorNode<K, V>>) thisNode.children()) {
```

Comment:
Do we need to iterate over children everytime we call forward(K, V, String)?
This doesn't sounds efficient since we can expect that this method will be called many times during processing.
How about introducing another field like `Map<String, ProcessorNode<?, ?>> nameToChild` inside ProcessorNode and a method like `ProcessorNode<?, ?> getChildByName(String name)` for much faster lookup?